### PR TITLE
docs(infrastructure-as-code.md): terraform isn't open source

### PR DESCRIPTION
**Pull request by xyhhx**
i also added a note about opentofu

> Before submitting the pull request, you must agree with the following statements by checking both boxes with a 'x'.
> * [x] "I accept that my contribution is placed under the CC BY-SA 2.0 license [1]."
> * [x] "My contribution complies with the Developer Certificate of Origin [2]." 
>
> [1] https://creativecommons.org/licenses/by-sa/2.0/
> [2] https://docs.xcp-ng.org/project/contributing/#developer-certificate-of-origin-dco

### CODE REVIEW

**Review by stormi — COMMENTED** [review]
Looks good to me

**Review by stormi — APPROVED** [review]

**Comment by xyhhx** [comment]
coming back to this, i realize that you can't use terraform's registry with opentofu, and vates hasn't submitted a provider to opentofu's registry. it's good you didn't merge this yet, but it should wait for that before being merged anyway.

i should probably update the doc to link to the new provider at that time, too

**Comment by stormi** [comment]
Thanks for your contribution!

**Pull request merged**
